packages/pipeline-editor/src/PipelineEditorWidget.tsx

```diff
@@ -163,33 +165,60 @@ export class PipelineEditor extends React.Component<
   widgetContext: DocumentRegistry.Context;
   position = 10;
   node: React.RefObject<HTMLDivElement>;
+  properties: any;
   propertiesInfo: any;
 
   constructor(props: any) {
     super(props);
     this.app = props.app;
     this.browserFactory = props.browserFactory;
-    this.canvasController = new CanvasController();
-    this.canvasController.setPipelineFlowPalette(palette);
     this.widgetContext = props.widgetContext;
-    this.widgetContext.ready.then(() => {
-      this.canvasController.setPipelineFlow(this.widgetContext.model.toJSON());
-    });
+
     this.toolbarMenuActionHandler = this.toolbarMenuActionHandler.bind(this);
     this.contextMenuHandler = this.contextMenuHandler.bind(this);
     this.contextMenuActionHandler = this.contextMenuActionHandler.bind(this);
     this.clickActionHandler = this.clickActionHandler.bind(this);
     this.editActionHandler = this.editActionHandler.bind(this);
     this.tipHandler = this.tipHandler.bind(this);
 
-    this.state = { showPropertiesDialog: false, propertiesInfo: {} };
+    // Read config from the backend
+    const palette: any = null;
+
+    Promise.resolve([
+      RequestHandler.makeGetRequest('elyra/pipeline/config/palette', false)
+    ])
+      .then(([palette]) => {
+        console.log(palette);
+      })
+      .catch((reason: Error) => {
+        console.error(reason.message);
+      });
 
-    this.initPropertiesInfo();
+    this.canvasController = new CanvasController();
+    this.canvasController.setPipelineFlowPalette(palette);
+    this.widgetContext.ready.then(() => {
+      this.canvasController.setPipelineFlow(this.widgetContext.model.toJSON());
+    });
+
+    this.state = { showPropertiesDialog: false, propertiesInfo: {} };
 
     this.applyPropertyChanges = this.applyPropertyChanges.bind(this);
     this.closePropertiesDialog = this.closePropertiesDialog.bind(this);
     this.openPropertiesDialog = this.openPropertiesDialog.bind(this);
 
+    Promise.resolve([
+      RequestHandler.makeGetRequest('elyra/pipeline/config/properties', false)
+    ])
+      .then(([properties]) => {
+        console.log(properties);
+        this.properties = properties;
+      })
+      .catch((reason: Error) => {
+        console.error(reason.message);
+      });
```

Comment:
```suggestion
    this.properties = await RequestHandler.makeGetRequest('elyra/pipeline/config/properties', false);
```

Reply:
I don't think you can do await in the constructor because it needs to be in an async method.
Maybe you can move the suggestions above to componentDidMount(), where
```
async componentDidMount(): Promise<void> {
  // request palette and properties config using await
}
```

Reply:
ah good point @karlaspuldaro , @lresende had to do that with an async call in #662